SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -8855,155 +8746,149 @@ void CEulerSolver::TurboAverageProcess(CSolver **solver, CGeometry *geometry, CC
 
 #endif
 
-    for (iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++){
-      for (iMarkerTP=1; iMarkerTP < config->GetnMarker_Turbomachinery()+1; iMarkerTP++){
+    /*--- Compute pitch-wise averaged quantities ---*/
+    for (auto iMarker = 0u; iMarker < config->GetnMarker_All(); iMarker++){
+      for (auto iMarkerTP=1; iMarkerTP < config->GetnMarker_Turbomachinery()+1; iMarkerTP++){
         if (config->GetMarker_All_Turbomachinery(iMarker) == iMarkerTP){
           if (config->GetMarker_All_TurbomachineryFlag(iMarker) == marker_flag){
 
-            TotalArea           = geometry->GetSpanArea(iMarker,iSpan);
-            AverageTurboNormal  = geometry->GetAverageTurboNormal(iMarker,iSpan);
-            nVert               = geometry->GetnTotVertexSpan(iMarker,iSpan);
+            auto TotalArea           = geometry->GetSpanArea(iMarker,iSpan);
+            auto AverageTurboNormal  = geometry->GetAverageTurboNormal(iMarker,iSpan);
+            auto nVert               = geometry->GetnTotVertexSpan(iMarker,iSpan);
 
             /*--- compute normal Mach number as a check for massflow average and mixedout average ---*/
             GetFluidModel()->SetTDState_Prho(TotalAreaPressure/TotalArea, TotalAreaDensity / TotalArea);
-            soundSpeed = GetFluidModel()->GetSoundSpeed();
-            MachTest   = TotalFluxes[0]/(TotalAreaDensity*soundSpeed);
+            auto soundSpeed = GetFluidModel()->GetSoundSpeed();
+            auto MachTest   = TotalFluxes[0]/(TotalAreaDensity*soundSpeed);
```

Comment:
We can't use auto for computations with su2double, it breaks AD

Reply:
Do you mean that  you always should define a variable as `su2double` or `const su2double` when the rhs is a computation?

Reply:
Yep https://su2code.github.io/docs_v7/Style-Guide/